Modules/itertoolsmodule.c

```diff
@@ -603,25 +604,25 @@ typedef struct {
 @classmethod
 itertools._grouper.__new__
 
-    parent: object(subclass_of='&groupby_type')
+    parent: object(subclass_of='clinic_state()->groupby_type')
     tgtkey: object
     /
 [clinic start generated code]*/
 
 static PyObject *
 itertools__grouper_impl(PyTypeObject *type, PyObject *parent,
                         PyObject *tgtkey)
-/*[clinic end generated code: output=462efb1cdebb5914 input=dc180d7771fc8c59]*/
+/*[clinic end generated code: output=462efb1cdebb5914 input=626b30a78e38cf7d]*/
 {
     return _grouper_create((groupbyobject*) parent, tgtkey);
 }
 
 static PyObject *
 _grouper_create(groupbyobject *parent, PyObject *tgtkey)
 {
-    _grouperobject *igo;
-
-    igo = PyObject_GC_New(_grouperobject, &_grouper_type);
+    PyTypeObject *tp = Py_TYPE(parent);
+    itertools_state *state = find_state_by_type(tp);
```

Comment:
An alternative approach here may be to store the state pointer in `groupbyobject`, thus avoiding `PyType_GetModuleByDef` here.

Reply:
That would be better that delivering a performance hit to itertools users.

Reply:
I agree; I'll update the PR. Thanks for your input, Raymond.